src/authz.js

```diff
@@ -1,40 +1,64 @@
 /**
  * The authorizer is responsible for taking User objects that only have
  * an identity set, and adding a collection of roles based on that
- * identity.  This is done by means of a set of plugins in the `authz`
- * directory and named in the configuration.
+ * identity.
  */
 class Authorizer {
   constructor(cfg) {
-    this. authorizers = cfg.app.authorizers.map((name) => {
-      return new (require('./authz/' + name))({cfg});
-    });
   }
 
   async setup() {
-    await Promise.all(this.authorizers.map(authz => authz.setup()));
   }
 
   /**
-   * Authorize the given user, calling user.addRole for all roles
-   * linked to user.identity.
+   * Authorize the given user.
+   * This method takes a User and attaches roles to it
    */
   async authorize(user) {
-    /* invoke all authorizers which recognize this identityProviderId */
-    return Promise.all(this.authorizers.map(authz => {
-      if (authz.identityProviders.indexOf(user.identityProviderId) !== -1) {
-        return authz.authorize(user);
+    const email = user.identityId;
+
+    // grant the everybody role to anyone who authenticates
+    user.addRole('everybody');
+
+    const mozGroupPrefix = 'mozilliansorg_';
+    const mozGroups = [];
+    const ldapGroups = [];
+
+    // Non-prefixed groups are what is known as Mozilla LDAP groups.
+    // Groups prefixed by a provider name and underscore are provided
+    // by a specific group engine. For example `providername_groupone`
+    // is provided by `providername`.
+    // Per https://goo.gl/bwWjvE
```

Comment:
There's nothing we can do about it, but it's worth noting that if IAM adds a new provider that we don't know about, we'll assume the results are LDAP groups.  For example, workday_contractors would be treated as an LDAP group, not a workday group.  I *think* this is harmless, as long as there aren't any name collisions -- and ensuring that is up to the IAM team.